packages/core/src/providers/netsuite.ts

```diff
@@ -0,0 +1,238 @@
+ *   providers: [
+ *       NetSuite({
+ *         clientId: NETSUITE_CLIENT_ID,
+ *         clientSecret: NETSUITE_CLIENT_SECRET,
+ *         accountId: NETSUITE_ACCOUNT_ID,
+ *        // Returns the current user using the N/runtime module. This url can be a suitelet or RESTlet (Recommended)
+ *        // Using getCurrentUser(); So we match this schema returned from this RESTlet in the profile callback. (Required)
+ *         userInfoUrl: "https://1234567.restlets.api.netsuite.com/app/site/hosting/restlet.nl?script=123&deploy=1",
+ *         profileCallback: (profile) => {
+ *           return {
+ *             id: profile.id,
+ *             name: profile.name,
+ *             email: profile.email,
+ *             location: profile.location,
+ *             role: profile.role,
+ *             contact: profile?.contact
+ *           }
+ *         },
+ *         // Optional
+ *         prompt: "login", // Required if you want to force the user to login every time.
+ *         scope: "restlets", // Optional defaults to "restlets rest_webservices". Enter the scope(s) you want to use followed by spaces.
+ *       })
+ *   ],
+ * })
+ * ```
+ *
+ * ### Resources
+ *
+ * - [NetSuite - Creating an Integration Record (OAuth 2.0)](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771733782.html#Related-Topics)
+ * - [NetSuite - Authorizing OAuth Requests](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps)
+ * - [NetSuite - Configure OAuth Roles](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_157771510070.html#Set-Up-OAuth-2.0-Roles)
+ * - [Learn more about NetSuite OAuth 2.0](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/chapter_157769826287.html#OAuth-2.0)
+ * - [Source code](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/netsuite.ts)
+ *
+ * ### Notes
+ *
+ * By default, Auth.js assumes that the NetSuite provider is
+ * based on the [OAuth 2](https://www.rfc-editor.org/rfc/rfc6749.html) specification.
+ *
+ * :::tip
+ * Make sure the `userInfoUrl` matches the return type of the profile callback to ensure the user session gets read correctly.
+ * To override the defaults for your use case, check out [customizing a built-in OAuth provider](https://authjs.dev/guides/providers/custom-provider#override-default-options).
+ *
+ * :::
+ *
+ * :::info **Disclaimer**
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function NetSuite(
+  config: OAuthNetSuiteOptions
+): string | object {

```

Comment:
Also this should return an `OAuthConfig` and take a `OAuthUserConfig` type object. You can import these from `.../providers/index.js`:

```ts
import type { OAuthConfig, OAuthUserConfig } from "./index.js"
```

You'll have to define `NetSuiteProfile` above somewhere, this shuold cover everythign NetSuite cuold return and a user might want to put on their profile. See `providers/gitlab.ts` for example.



```suggestion
export default function NetSuite<P extends NetSuiteProfile>(
  options: OAuthUserConfig<P>
): OAuthConfig<P> {
```